src/libraries/Microsoft.Extensions.Logging.TraceSource/src/TraceSourceLogger.cs

```diff
@@ -33,15 +35,18 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                 {
                     message += state;
                 }
-                if (exception != null)
-                {
-                    message += Environment.NewLine + exception;
-                }
```

Comment:
- [ ] Why have Environment.NewLine as exception delimiter and not just a single whitespace? " ".
Using single whitespace would allow to keep having the log in a single line for those expected like this until now.

(If you compare this with Console logging, you would see that JsonConsoleFormatter and SimpleConsoleFormatter give the option to either have new line or a single whitespace, but SystemdConsoleFormatter uses single whitespace as delimiter). I would vote to have single whitespace here to make it that bit more consistent with console logging.

- [ ] Even though it becomes more readable to add "Error: " in:
```
message += exceptionDelimiter + "Error: " + exception;
```

I don't see this elsewhere, how about we perhaps stick with below instead?
```c#
message += exceptionDelimiter + exception;
```

- [ ] Nit: now that you removed exception null check in else block, how about from:
```c#
            else
            {
                if (exception != null)
                {
                    message += Environment.NewLine + exception;
                }
            }
```
to 
```c#
            else if (exception != null)
            {
                 message += Environment.NewLine + exception;
            }
```

- [ ] Nit: just to keep consistent with surrounding code blocks perhaps you could add:
```diff
            if (string.IsNullOrEmpty(message) && exception == null)
+           {
                return;
+           {
```

Reply:
In summary, recommending:

```c#
            if (!IsEnabled(logLevel))
            {
                return;
            }
            string message = string.Empty;
            if (formatter != null)
            {
                message = formatter(state, exception);
            }
            else if (state != null)
            {
                message += state;
            }

            if (exception != null)
            {
                string exceptionDelimiter = string.IsNullOrEmpty(message) ? string.Empty : " ";
                message += exceptionDelimiter + exception.ToString();
            }
            if (!string.IsNullOrEmpty(message))
            {
                _traceSource.TraceEvent(GetEventType(logLevel), eventId.Id, message);
            }
```

And for sample:
```c#
        static void Main(string[] args)
        {
            using var loggerFactory = LoggerFactory.Create(builder =>
            {
                var testSwitch = new SourceSwitch("sourceSwitch", "Logging Sample");
                testSwitch.Level = SourceLevels.Warning;
            builder
                .AddSystemdConsole()
                .AddTraceSource(testSwitch,
                    new TextWriterTraceListener(@"C:\CodeHub\logs\trace.log"));// writer: Console.Out));
            });
            var logger = loggerFactory.CreateLogger("sourceSwitch");
            logger.Log(LogLevel.Warning, new NotImplementedException("NIE exception message"), "message for warning");
        }
```

Would get a single line log, rather than multiple line for a log:
```
sourceSwitch Warning: 0 : message for warning System.NotImplementedException: NIE exception message
```
Which is similar to SystemdConsoleFormatter output.

Reply:
hi, thx for your positive feedback, I absolutely agree with you, next time I try to compare my code to similar implementations
br